org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/healthcheck/AbstractHealthCheckServlet.java

```diff
@@ -114,6 +114,10 @@ protected void doChecks(RunContext context, HttpServletRequest req, HttpServletR
       resp.getWriter().print(Platform.get().inDevelopmentMode() ? detailedOutput : output);
     }
     else {
+      if (statusCode != HttpServletResponse.SC_OK) {
+        String detailedOutput = generateOutput(statusCode, checks, failed, true);
+        LOG.warn(detailedOutput);
+      }
```

Comment:
Definitely better than before, just one comment:

Do we really want to print a warning only if level is higher than _DEBUG_? If _DEBUG_ level is enabled, it would just be printed out as debug log. I would suggest always printing a warning if status does not equal 200.

For example (for the whole block):

```
    LazyValue<String> detailedOutput = new LazyValue<>(() -> generateOutput(statusCode, checks, failed, true));
    boolean isDevelopmentMode = Platform.get().inDevelopmentMode();
    if (statusCode != HttpServletResponse.SC_OK) {
      LOG.warn(detailedOutput.get());
    }
    else if (LOG.isDebugEnabled() || isDevelopmentMode) {
      LOG.debug(detailedOutput.get());
    }
    resp.getWriter().print(isDevelopmentMode ? detailedOutput.get() : output);
```

What do you think?

Reply:
Good idea, I use your code now.